Added two workflows and .editorconfig

SeniorMars opened the pull request:
The workflows are:
    - dependabot
    - check

Dependabot just checks that dependencies are safe and upgraded if needed.

Check adds a bunch of things I feel are needed.

Note: Both of these workflows come from: https://github.com/jonhoo/rust-ci-conf

An .EditorConfig is provide as it is very important to have the same style everywhere regardless of editor.

SeniorMars commented:
Note: check clippy fails right here because this pr is not yet merged as such doesn't have permissions.

SeniorMars commented:
Also, you should probably turn off allowing workflows to run if there are from public forks: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

laurmaedje commented:
I like how light the CI is currently. Personally, I also don't really like dependabot and clippy. The two things I would be fine with is checking formatting and docs. I think these could be integrated into the existing `ci.yml` workflow.

SeniorMars commented:
I deleted the clippy and hack job and added the rest to `ci.yml`. I kept `msrv` (minimal supported version) as I feel it is useful. Please let me know if there is anything else I can change.

drupol commented on .github/workflows/ci.yml:
```diff
@@ -10,3 +15,47 @@ jobs:
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - name: cargo fmt --check
```
How about using a name instead of using the command here?

drupol commented on .github/workflows/ci.yml:
```diff
@@ -10,3 +15,47 @@ jobs:
+          submodules: true
+      - name: Install nightly
+        uses: dtolnay/rust-toolchain@nightly
+      - name: cargo doc
```
`Build doc` instead ?

drupol reviewed (commented):
Added minor syntactic comments

laurmaedje commented:
I added a formatting check to CI and an editor configuration.
